src/trace.ml

```diff
@@ -55,9 +55,15 @@ let evaluate_symbol_selection ~symbol_selection ~elf ~header =
   match symbol_selection with
   | Symbol_selection.Use_fzf_to_select_one ->
     let all_symbol_names = Elf.all_symbols elf |> List.map ~f:Tuple2.get1 in
+    let demangled_symbols =
+      List.map all_symbol_names ~f:(fun mangled_symbol ->
+          match Demangle_ocaml_symbols.demangle mangled_symbol with
+          | None -> mangled_symbol
+          | Some demangled_symbol -> demangled_symbol)
+    in
     if force supports_fzf
     then (
-      match%bind Fzf.pick_one ~header (Inputs all_symbol_names) with
+      match%bind Fzf.pick_one ~header (Inputs demangled_symbols) with
```

Comment:
@ClyveGassant Tudor points out that this will break magic-trace, because magic-trace will attempt to attach to the _demangled_ symbol. We should keep a mapping (maybe Fzf library already has this feature) back to the original, mangled symbol, probably. Let's talk on Monday!

Reply:
Hi Hao! Thank you for the comment! I have fixed the problem and pushed the code.